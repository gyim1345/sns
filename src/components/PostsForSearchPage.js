import React from "react";
import { Link } from "react-router-dom";

function PostsForSearchPage({ posting, setUserOfActivePage }) {
  const changeUser = () => {
    setUserOfActivePage(posting.userName);
  };
  return (
    <>
      <Link
        to={`/posting/${posting.id}`}
        onClick={changeUser}
      >
        <div className="test" style={{ position: "relative", width: "26vw" }}>
          <div
            style={{
              position: "absolute",
              top: 0,
              left: 0,
              bottom: 0,
              right: 0,
              backgroundImage: `url(${posting.imageUrl})`,
              backgroundSize: "cover",
              backgroundPosition: "center"
            }}
          ></div>
          <div style={{ paddingTop: "100%" }}></div>
        </div>
      </Link>
    </>
  );
}

export default PostsForSearchPage;
