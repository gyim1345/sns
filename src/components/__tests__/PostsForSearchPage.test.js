import React from 'react';
import { mount } from 'enzyme';
import PostsForSearchPage from '../PostsForSearchPage';
import { post, currentUser } from './testjs';
import { MemoryRouter } from 'react-router-dom';

describe('<PostsForSearchPage />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/1', key: 'testKey' }]}>
        <PostsForSearchPage posting={post} setUserOfActivePage={currentUser} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
