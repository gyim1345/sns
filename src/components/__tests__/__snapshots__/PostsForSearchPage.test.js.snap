// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsForSearchPage /> 스냅샷 비교 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/1",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/1",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/1",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PostsForSearchPage
      posting={
        Object {
          "_id": "5e8ec52d9bb4913b5a844423",
          "id": 1,
          "imageUrl": "http://localhost:3000/static/images/1.jpg",
          "like": Array [
            "gibong@gmail.com",
          ],
          "tag": Array [
            "landscape",
          ],
          "title": "posting with id 1",
          "userName": "gibong@gmail.com",
        }
      }
      setUserOfActivePage="gibong@gmail.com"
    >
      <Link
        onClick={[Function]}
        to="/posting/1"
      >
        <LinkAnchor
          href="/posting/1"
          navigate={[Function]}
          onClick={[Function]}
        >
          <a
            href="/posting/1"
            onClick={[Function]}
          >
            <div
              className="test"
              style={
                Object {
                  "position": "relative",
                  "width": "26vw",
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundImage": "url(http://localhost:3000/static/images/1.jpg)",
                    "backgroundPosition": "center",
                    "backgroundSize": "cover",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <div
                style={
                  Object {
                    "paddingTop": "100%",
                  }
                }
              />
            </div>
          </a>
        </LinkAnchor>
      </Link>
    </PostsForSearchPage>
  </Router>
</MemoryRouter>
`;
